net/net_evbuffer.c

```diff
@@ -1551,11 +1561,45 @@ static int process_payload(struct event_info *e)
     }
 }
 
+static void evbuffer_free_callback(const void *ptr, size_t unused0, void *unused1)
+{
+    free((void *)ptr);
+}
+
 static ssize_t readv_evbuffer(struct evbuffer *buf, int fd)
 {
 #   define NVEC 8
     struct iovec v[NVEC];
-    const int nv = evbuffer_reserve_space(buf, MB(4), v, NVEC);
+    int avail = MB(4);
+
+    /*
```

Comment:
The 4 lines in ifdef is pretty self evident. Don't think we need this wall of text anymore.